### DISCUSSION TITLE
Arc's command bar for Zen browser

### DISCUSSION THREAD

**Discussion by danielbolivar** [zen-browser/desktop#1116] [Ideas]
I would love to see Zens command bar implemented in Zen browser. Maybe a rough first implementation could be just displaying the bar for normal internet queries. Something like just a hotkey that trigger the [floating url bar](https://github.com/zen-browser/theme-components/tree/main/floating-urlbar)

![image](https://github.com/user-attachments/assets/69883f5c-4413-4ce0-bb17-c1773a441530)

**Comment by giuli635** [comment]
You can trigger the bar using Ctrl+L

**Reply by danielbolivar** [reply]
Yeah but it also triggers when you tap the urlbar. The point is it just happens when you click the shortcut. This is because it could be sometimes annoying when you are not expecting it to pop

**Reply by giuli635** [reply]
Ooooh, yes, I see what you mean. Yes it's a bit annoying because you have to move the mouse from the position where the url bar was.

**Reply by Tealk** [reply]
This only works to a limited extent for me, first the url is highlighted in the address bar and then when I start typing the floating window appears.

**Comment by Kevoin** [comment]
It would be nice to be able to perform site search. On arc you can open the command bar, type "y" and the tab to go directly to youtube search. This works for multiple websites, not just youtube, and is such a nice workflow improvement and keeps you from getting distracted😉

**Reply by elianiva** [reply]
AFAIK you can already do this kind of shortcut, you just need to add the trigger in the settings

the difference is it's in the url bar instead of the command bar

**Comment by shaqaruden** [comment]
I would like to see the ability to switch tabs from the address bar to be improved. It can take too many keyboard inputs after after typing in my "query" to be able to switch tabs. For example

If I type "Youtube" and I have a single opened tab for YouTube I should be able to Press the up arrow or down arrow to highlight to switch to that tab. I just tried this now and I either have to down arrow through 15+ results or arrow up through the "This time, search with" section before I can get to the option to switch to that tab

**Comment by Dedsecproject** [comment]
I'm looking for the ability to trigger the features that Zen browser has through the command bar instead of searching through the settings to find the shortcut needed for that feature to be triggered. I Would love to see that you can type in on the command bar what you want. For example, you type in "split" and the option shows up displaying which shortcut is linked to said feature.

![SCR-20241121-ithn](https://github.com/user-attachments/assets/60c39be1-5ccf-40f3-8c13-b81328bfabc5)

**Reply by baggiponte** [reply]
I absolutely need this feature 🤣 That's my favorite thing about arc right after the sidebar tabs.

**Reply by LaserKaspar** [reply]
Does this feature exist in zen?

**Reply by Dedsecproject** [reply]
not at the moment

**Reply by adminy** [reply]
would be pretty nice to see this feature, because its not obvious what the shortcuts are and this could be a very good way to learn them.

**Reply by aSapien** [reply]
Absolutely need this feature. I am often using Arc's "Move to workspace" commands and the like

**Reply by louis-bourgault** [reply]
Or have it similar to VS code -- like how > brings it up or ctrl shift p. I keep hitting that shortcut hoping and it just opens a private window instead 😞

**Comment by md-tanveer** [comment]
Hope to see the commend bar in the future updates

**Comment by tobias-hay** [comment]
Bumping this thread, would love to be able to do all my searches without having to open a "new tab" window every time, much cleaner to have an option to have it like arcs.

**Comment by svliantiss** [comment]
How can you change the CMD + L to CMD + T, that should/would also fix opening a new tab before entering the url?

**Comment by vraravam** [comment]
imo, any application (in general) that needs this kind of capability, will need to revamp (not commenting whether its a major or a minor change) to refactor all menu items and context menu items to be bound/tied to actions, and then the actions themselves carry out the actual intended work. This allows for key bindings to be applied/associated to these actions.
I don't think FF exposes this abstract notion of an action except for a few capabilities. Till such time that FF does this refactoring, I don't think Zen can get such a capability. I might be totally wrong, would love to be corrected.

**Comment by viwaswatraj** [comment]
Since most of my workflow involves keeping my hands on the keyboard, this proposed feature would significantly boost productivity.
I would like to map certain keywords to accomplish certain tasks. For example, typing `st`  in the command bar (for [S]witch [T]abs) followed by a tab key would allow me to see all open tabs and search through them either via query or by scrolling.

**Comment by louis-bourgault** [comment]
For the command bar: 
It looks like this kind of command in a limited way has been added to Firefox, so there definitely is a framework for it in the newest version! I was playing around with another machine with FF yesterday and there's now things like a 'print' shortcut from the tab bar. In the release notes for firefox 137.0:

> Secondary Action Buttons: Suggestions for common Firefox features.
https://www.mozilla.org/en-US/firefox/137.0/releasenotes/

It'd need to be extended for Zen of course to add space switching stuff but definitely could work

**Comment by shaqaruden** [comment]
One issue I see with the current feature set of a "command bar" that Zen does have is, if a tab is unloaded I cannot switch to it and load it. Instead a new tab is created.
